PyRDF/backend/Local.py

```diff
@@ -58,7 +58,9 @@ def execute(self, generator):
         """
         mapper = generator.get_callable()  # Get the callable
 
-        if not self.pyroot_rdf:
+        # if the RDataFrame has not been created yet or if a new one
+        # is created by the user in the same session
+        if (not self.pyroot_rdf) or (self.pyroot_rdf != generator.head_node):
```

Comment:
I would use:

```suggestion
        if (not self.pyroot_rdf) or (self.pyroot_rdf is not generator.head_node):
```

To check the identity rather than the equality, although probably they will never be the same.